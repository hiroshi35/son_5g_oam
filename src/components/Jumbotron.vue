<template>
  <div :class="'jumbotron ' + className" :style="style">
    <div class="container">
      <h1>{{title}}</h1>
      <p>
        <slot></slot>
      </p>
      <p>
        <slot name="buttons"></slot>
      </p>
    </div>
  </div>
</template>

<script>
  export default {
    props: {
      title: {
        type: String,
        required: true
      },
      img: {
        type: String
      },
      className: {
        type: String,
        default: ''
      }
    },
    computed: {
      style () {
        if (this.img) {
          return {
            backgroundImage: 'url(' + this.img + ')',
            color: '#fff',
            backgroundRepeat: 'no-repeat',
            backgroundSize: 'cover'
          }
        }
        return ''
      }
    }
  }
</script>
