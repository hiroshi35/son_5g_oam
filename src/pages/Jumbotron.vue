<template>
  <div id="page-wrapper">

    <jumbotron title="Hello World" img="/static/imgs/pexels-photo-235615.jpeg">
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad cumque dolore eligendi error, esse incidunt ipsa magni, minima molestias mollitia natus quae saepe sapiente sed similique voluptas voluptate. Impedit, iste.</p>
      <p slot="buttons">
        <button class="btn btn-outlined-orange rounded">Learn More</button>
      </p>
    </jumbotron>
    <div class="container-fluid">

      <div class="row">
        <div class="col-md-6">
          <panel type="panel-white right-angular">
            <h3 slot="header">Default Jumbotron</h3>
            <jumbotron title="Hello World">
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad cumque dolore eligendi error, esse incidunt ipsa magni, minima molestias mollitia natus quae saepe sapiente sed similique voluptas voluptate. Impedit, iste.</p>
              <p slot="buttons">
                <button class="btn btn-outlined-blue right-angular">Learn More</button>
              </p>
            </jumbotron>

          </panel>
        </div>
        <div class="col-md-6">
          <panel type="panel-white right-angular">
            <h3 slot="header">Jumbotron with Image</h3>
            <jumbotron title="Hello World" img="/static/imgs/pexels-photo-459059.jpeg">
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad cumque dolore eligendi error, esse incidunt ipsa magni, minima molestias mollitia natus quae saepe sapiente sed similique voluptas voluptate. Impedit, iste.</p>
              <p slot="buttons">
                <button class="btn btn-outlined-teal right-angular">Learn More</button>
              </p>
            </jumbotron>

          </panel>
        </div>
      </div>
    </div>
    <!-- /.container-fluid -->


  </div>
</template>

<script>
  import Jumbotron from '../components/Jumbotron.vue'

  export default {
    components: {
      Jumbotron
    }
  }
</script>