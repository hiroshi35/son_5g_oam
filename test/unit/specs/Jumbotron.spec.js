import { shallow } from 'vue-test-utils'
import Jumbotron from '@/components/Jumbotron.vue'

describe('Jumbotron.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Jumbotron,
      {
        propsData: {
          title: 'Hello World',
          img: '/static/imgs/carousel/1.jpg'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has correct props', () => {
    expect(wrapper.props().title).toBe('Hello World')
    expect(wrapper.props().img).toBe('/static/imgs/carousel/1.jpg')
  })
})
