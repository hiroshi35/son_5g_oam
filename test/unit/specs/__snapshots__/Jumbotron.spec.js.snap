// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbotron.vue has the expected html structure 1`] = `
<div
  class="jumbotron "
  style="background-image: url(/static/imgs/carousel/1.jpg); color: rgb(255, 255, 255); background-repeat: no-repeat; background-size: cover;"
>
  <div
    class="container"
  >
    <h1>
      Hello World
    </h1>
     
    <p />
     
    <p />
  </div>
</div>
`;
